iceoryx_posh/include/iceoryx_posh/version/compatibility_check_level.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_POSH_VERSION_COMPATIBILITY_CHECK_LEVEL_HPP
+#define IOX_POSH_VERSION_COMPATIBILITY_CHECK_LEVEL_HPP
+
+namespace iox
+{
+namespace version
+{
+enum class CompatibilityCheckLevel
```

Comment:
Check is redundant in this name.

Reply:
I want to do a compatibility check and I need a level to define "how detailed is the compatibility check". Which name do you prefer?

Reply:
I'm with @ithier and wouldn't use `Check`. It's the level we want to be compatible with, we're not checking anything here. That should probably also involve `m_compatibilityCheckLevel`. I know, quite some code changes.